js/components/common/Disclaimer.js

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+
+/**
+ * A React component representing the disclaimer popup
+ */
+export default class Disclaimer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            dontShowDisclaimer: localStorage.getItem('DontShowDisclaimer')
```

Comment:
The names here could be better. The state name should be something standard like `hidden`, and the local storage name a corresponding `hide-disclaimer`.

Reply:
Oh also, do the conversion to a boolean here rather than in the `render`. Logically, this state should be a bool, not a string.